o2a/mappers/ssh_mapper.py

```diff
@@ -37,48 +38,68 @@ def __init__(
         self,
         oozie_node: Element,
         name: str,
+        job_properties: Dict[str, str],
+        configuration_properties: Dict[str, str],
         trigger_rule: str = TriggerRule.ALL_SUCCESS,
-        params: Dict[str, str] = None,
         template: str = "ssh.tpl",
         **kwargs,
     ):
-        ActionMapper.__init__(self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, **kwargs)
-
-        if params is None:
-            params = {}
+        ActionMapper.__init__(
+            self,
+            oozie_node=oozie_node,
+            name=name,
+            trigger_rule=trigger_rule,
+            job_properties=job_properties,
+            configuration_properties=configuration_properties,
+            **kwargs,
+        )
         self.template = template
+        cmd = self.get_command()
+
+        self.command = el_utils.convert_el_to_jinja(cmd)
+        host_key = self.get_host_key(self.property_set)
+
+        # Since Airflow separates user and host, we can't use jinja templating.
+        # We must check if it is in job_properties.
+        user_host = host_key.split("@")
+        self.user = user_host[0]
+        self.host = user_host[1]
 
+    def get_command(self) -> str:
         cmd_node = self.oozie_node.find("command")
         arg_nodes = self.oozie_node.findall("args")
         if cmd_node is None or not cmd_node.text:
             raise Exception("Missing or empty command node in SSH action {}".format(self.oozie_node))
         cmd = cmd_node.text
         args = (x.text if x.text else "" for x in arg_nodes)
         cmd = " ".join(shlex.quote(x) for x in [cmd, *args])
+        return cmd
 
-        self.command = el_utils.convert_el_to_jinja(cmd, quote=True)
+    def get_host_key(self, property_set: PropertySet) -> str:
```

Comment:
Is this argument necessary? I think this method can refer to this object directly.

Reply:
Indeed. Fixed in 46e3cd5b8e2835bc191622530affa0f5a5e54a0b